Tasks/Common/webdeployment-common/kuduutility.ts

```diff
@@ -157,31 +162,158 @@ async function uploadFiletoKudu(publishUrl: string, physicalPath: string, header
+    httpObj.get('GET', kuduGetFileUrl, headers, (error, response, body) => {
+        if(error) {
+            defer.reject({ error: error + ''});
+        }
+        else if(response.statusCode === 200) {
+            tl.debug('retrieved file content : ' + fileName);
+            defer.resolve({
+                content: body
+            });
+        }
+        else {
+            defer.reject(
+                {
+                    statusCode: response.statusCode,
+                    statusMessage: response.statusMessage 
+                }
+            );
+        }
+    });
+    return defer.promise;
+}
+
+async function pollForFile(publishingProfile, physicalPath, fileName) {
+    var defer = Q.defer();
+    var attempts = 0;
+    tl.debug('Polling started for file: ' + fileName);
+    var poll = async function () {
+        if (attempts < 180) {
+            attempts += 1;
+            var fileContent;
+            try {
+                fileContent = await getFileContent(publishingProfile, physicalPath, fileName);
+                tl.debug('Found file:  ' + fileName);
+                defer.resolve('');
+            }
+            catch(error) {
+                if(error.statusCode === 404) {
+                    tl.debug('File ' + fileName + ' not found. rerty after 10 seconds. Attempt : ' + attempts);
+                    setTimeout(poll, 10000);
+                }
+                else {
+                    if(error.error) {
+                        defer.reject(tl.loc('FailedToGetKuduFileContentError', fileName, error.error));
+                    }
+                    else {
+                        defer.reject(tl.loc('FailedToGetKuduFileContent', fileName, error.statusCode, error.statusMessage));
+                    }
+                }
+            }
+        }
+        else {
+            defer.reject(tl.loc('ScriptStatusTimeout'));
+        }
+    }
+    poll();
+    return defer.promise;
+}
+
+async function getPostDeploymentScriptLogs(publishingProfile, physicalPath, uniqueID) {
+    var basicAuthToken = 'Basic ' + new Buffer(publishingProfile.userName + ':' + publishingProfile.userPWD).toString('base64');
+    var headers = {
+        'Authorization': basicAuthToken,
+        'If-Match': '*'
+    };
+    try {
+        var stdoutLog = (await getFileContentUtility(publishingProfile, physicalPath, 'stdout_' + uniqueID + '.txt'));
+        var stderrLog = (await getFileContentUtility(publishingProfile, physicalPath, 'stderr_' + uniqueID + '.txt'));
+        var scriptReturnCode = (await getFileContentUtility(publishingProfile, physicalPath, 'script_result_' + uniqueID + '.txt')).trim();
+    }
+    catch(error) {
+        throw Error(error);
+    }
+    if(stdoutLog) {
+        console.log(tl.loc('stdoutFromScript'));
+        console.log(stdoutLog);
+    }
+    if(stderrLog) {
+        console.log(tl.loc('stderrFromScript'));
+        console.log(stderrLog);
```

Comment:
If you are anyway throwing the stderrLog, why do you display in console as well?